finagle-backend/src/main/scala/sttp/client/finagle/FinagleBackend.scala

```diff
@@ -0,0 +1,165 @@
+    case Method.POST    => FMethod.Post
+    case Method.PUT     => FMethod.Put
+    case Method.DELETE  => FMethod.Delete
+    case Method.OPTIONS => FMethod.Options
+    case Method.PATCH   => FMethod.Patch
+    case Method.CONNECT => FMethod.Connect
+    case Method.TRACE   => FMethod.Trace
+    case _              => FMethod(m.method)
+  }
+
+  def requestBodyToFinagle(r: Request[_, Nothing]): http.Request = {
+    val finagleMethod = methodToFinagle(r.method)
+    val url = r.uri.toString
+    val headers = headersToMap(r.headers)
+
+    r.body match {
+      case FileBody(f, _) =>
+        val content : String = Source.fromFile(f.toFile).mkString
+        buildRequest(url, headers, finagleMethod, Some(ByteArray(content.getBytes: _*)))
+      case NoBody => buildRequest(url, headers, finagleMethod, None)
+      case StringBody(s, e, _) => buildRequest(url, headers, finagleMethod, Some(ByteArray(s.getBytes(e): _*)))
+      case ByteArrayBody(b, _) => buildRequest(url, headers, finagleMethod, Some(ByteArray(b: _*)))
+      case ByteBufferBody(b, _) => buildRequest(url, headers, finagleMethod, Some(ByteBuffer.Owned(b)))
+      case InputStreamBody(is, _) => buildRequest(url, headers, finagleMethod, Some(ByteArray(Stream.continually(is.read).takeWhile(_ != -1).map(_.toByte).toArray: _*)))
+      case MultipartBody(parts) => val requestBuilder = RequestBuilder.create().url(r.uri.toString).addHeaders(headersToMap(r.headers))
+        val elements = parts.map {
+        part =>
+          getBasicBodyContent(part)
+        }
+        requestBuilder.add(elements).buildFormPost(true)
+        //requestBuilder.addFormElement(elements: _*).buildFormPost(true)
+      case _ => buildRequest(url, headers, finagleMethod, None)
+    }
+  }
+
+  def getBasicBodyContent(part: Part[BasicRequestBody]): FormElement = {
+
+    val content: String = part.body match {
+      case StringBody(s, _, _) => s
+      case ByteArrayBody(b, _) => Source.fromBytes(b).mkString
+      case ByteBufferBody(b, _) => Source.fromBytes(b.array()).mkString
+      case InputStreamBody(is, _) => Source.fromInputStream(is).mkString
+      case FileBody(f, _) => Source.fromFile(f.toFile).mkString
+    }
+
+    part.fileName match {
+      case Some(_) => FileElement(part.name, ByteArray(content.getBytes: _*), part.contentType, part.fileName)
+      case None  => SimpleElement(part.name, content)
+    }
+
+  }
+  def buildRequest(url: String, headers: Map[String, String], method: FMethod, content: Option[Buf]): http.Request = {
+    RequestBuilder.create().url(url).addHeaders(headers).build(method, content)
+  }
+
+  def fromFinagleResponse[T](rr: ResponseAs[T, Nothing],  r: FResponse, meta: ResponseMetadata): TFuture[T] = {
+
+
+    rr match {
+      case MappedResponseAs(raw, g) =>
+        fromFinagleResponse(raw, r, meta).map(h => g(h,meta))
+
+      case ResponseAsFromMetadata(f) => fromFinagleResponse(f(meta), r, meta)
+
+      case IgnoreResponse =>
+        TFuture.Done
```

Comment:
don't we have to read the response's body content anyway? so that the connection can be later released (that's often the case in web clients)

Reply:
or maybe finagle has a way of draining the response's body

Reply:
there is no method to drain the body except to set the content buffer to `Empty`. Will change it to that